### PR TITLE
Enable eager spmd

### PR DESCRIPTION
**Pull request by JackCaoG**

### CODE REVIEW

**Comment by alanwaketan** [comment]
How could SPMD possibly work for eager mode?

**Comment by JackCaoG** [comment]
> How could SPMD possibly work for eager mode?

consider eager mode as calling `mark_step` after every pytorch op.

**Comment by alanwaketan** [comment]
> > How could SPMD possibly work for eager mode?
> 
> consider eager mode as calling `mark_step` after every pytorch op.

Then how sharding propogation and auto partition work? I assume they don't carry states from last graph?

**Comment by alanwaketan** [comment]
Okay, that's fair.

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -2742,7 +2742,9 @@ void XLANativeFunctions::_propagate_xla_data(const at::Tensor& input,
 
   // 2) Aid SPMD.
   XLATensor::ShardingSpecPtr sharding = input_tensor->sharding_spec();
-  if (sharding && sharding->sharding.type() != xla::OpSharding::UNKNOWN) {
+  // don't propagate sharding in eager mode.
+  if (!XLAGraphExecutor::Get()->UseEagerMode() && sharding &&
```
Comment (alanwaketan) [inline torch_xla/csrc/aten_xla_type.cpp:2746]: May I ask why?

Comment (JackCaoG) [reply torch_xla/csrc/aten_xla_type.cpp:2746]: It complained about the output tensor already has a sharding and we can't propagate to it. This happens in the backward. I didn't spend enough time to debug it but I don't expect user to actually run eager mode with step fn(forward and backward), I only expect them to run it with some data preprocessing on device so I just quickly unblock myself.

**Review by alanwaketan — APPROVED** [review]
LGTM. Thanks, Jack!

**Pull request merged**